pkg/tsdb/azuremonitor/aztokenprovider/token_provider.go

```diff
@@ -93,9 +98,9 @@ func (provider *tokenProviderImpl) resolveAuthorityHost(cloudName string) string
 		return azidentity.AzureGovernment
 	case setting.AzureGermany:
 		return azidentity.AzureGermany
+	default:
+		return ""
 	}
-	// Fallback to direct URL
-	return provider.authParams.Url
```

Comment:
why this change?

Reply:
While we didn't actually use it, I initially implemented that  `authParams.Url` was used to provide custom IdP authorization endpoint, e.g. for unknown cloud or for testing. But it was implemented the way that value from `authParams.Url` was used only if cloud didn't match.

In the new implementation `AzureClientSecretCredentials` struct has fields `AzureCloud` and `Authority`:

```go
type AzureClientSecretCredentials struct {
	AzureCloud   string
	Authority    string
	...
}
```

If `Authority` isn't set (as is the normal case) then default authority used for a given cloud `AzureCloud`.
If `Authority` is set, then it overrides default authority from the cloud.